ui-admin/src/study/notifications/EmailTemplateEditor.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useRef, useState } from 'react'
+import EmailEditor, { EditorRef, EmailEditorProps } from 'react-email-editor'
+import { EmailTemplate } from '@juniper/ui-core'
+import { Tab, Tabs } from 'react-bootstrap'
+import { getImageBaseUrl } from 'api/api'
+
+/** Enables editing an email with design/preview modes */
+export default function EmailTemplateEditor({ emailTemplate, updateEmailTemplate, portalShortcode }: {
+    emailTemplate: EmailTemplate, portalShortcode: string, updateEmailTemplate: (emailTemplate: EmailTemplate) => void
+}) {
+  const emailEditorRef = useRef<EditorRef>(null)
+  const [activeTab, setActiveTab] = useState<string | null>('designer')
+
+  const replacePlaceholders = (html: string) => {
+    return html.replaceAll('${siteImageBaseUrl}', location.origin + getImageBaseUrl(portalShortcode))
+  }
+  const insertPlaceholders = (html: string) => {
+    return html.replaceAll(location.origin + getImageBaseUrl(portalShortcode), '${siteImageBaseUrl}')
+  }
+
+  const onEditorLoaded: EmailEditorProps['onReady'] = unlayer => {
+    unlayer.loadDesign({
+      // @ts-ignore
+      html: replacePlaceholders(emailTemplate.body),
+      classic: true
+    })
+    unlayer.addEventListener('design:updated', () => {
+      if (!emailEditorRef.current?.editor) { return }
+      emailEditorRef.current.editor.exportHtml(data => {
+        updateEmailTemplate({
+          ...emailTemplate,
+          body: insertPlaceholders(data.html)
+        })
+      })
+    })
+  }
+
+  const templateVersionString = `v${emailTemplate.version}`
+  return <div className="mt-3">
+    <div className="d-flex align-items-baseline">
+      <h3 className="h6">{emailTemplate.name}</h3>
+      <div className="ms-2 text-muted fst-italic">
+                ({emailTemplate.stableId} {templateVersionString})
+      </div>
+    </div>
+    <div>
+      <label className="form-label">Subject
+        <input className="form-control" type="text" size={100} value={emailTemplate.subject}
+          onChange={e => updateEmailTemplate({
+            ...emailTemplate,
+            subject: e.target.value
+          })}/>
```

Comment:
Testing note: editing the subject, and then editing the body in the Designer tab will cause the subject to be reset to the original value (you'll need to click somewhere else on the page to trigger the event listener)

Reply:
oooh, that's weird, investigating now